src/octokit-helper.ts

```diff
@@ -0,0 +1,23 @@
+import * as github from '@actions/github'
+import {Octokit} from '@octokit/rest'
+import {getServerApiUrl} from './url-helper'
+
+// Centralize all Octokit references by re-exporting
+export {Octokit} from '@octokit/rest'
+
+export type OctokitOptions = {
+  baseUrl?: string
+  userAgent?: string
+}
+
+export function getOctokit(authToken: string, opts: OctokitOptions) {
+  const options: Octokit.Options = {
+    baseUrl: getServerApiUrl(opts.baseUrl)
```

Comment:
can we also conditionally set the baseUrl when the `opts.baseUrl` is provided?

Reply:
That is technically what the function that is being called is doing, as it replicates the octokit default behaviour of a fallback to github.com/api, https://github.com/actions/checkout/pull/922/files#diff-f53c8d0b0ecfe6c87df94f6060fbc83aa0ec2c24f279c8025be59febb6a68dc0R22

it is possible to set this conditionally, but it is less obvious that this is the behaviour as the default in octokit is a separate library and buried deeper in the code base.

Reply:
Understand, my view was for safe rollout and minimizing any risk.
We were relying on `@actions/github` to populate the default baseUrl, in case there is any change in how to populate the baseUrl in the default case, upgrading `@actions/github` to the latest will handle it for us.

Reply:
I can indeed understand that. This is not the only place where you have defaults like this.
The way that I implemented this was to consolidate the already default fallback nature in a way that is consistent for the api and host, as the code base has an implicit fallback on api.github.com vs the GHES equivalent and one on a GHES server base url or https://github.com. To properly connect these decisions together as one is defaulted in this code base and the other is hurried layers deep in the octokit wrapper, that is why I did it this way.

Sure it can be done to utilize the fallback, but then you have one assumption here in this action for the git commands and another burried deep inside octokit that operates on a default, that in all honesty is never used as the environment URLs are already populated and used as fallbacks before you get that empty behaviour in actions in the first place.

I am happy to refactor this again if it is perceived to be a blocking factor in this change, but I think in all honesty it removes clarity and will make the changes for injection potentially more messy than what we have, as well as leading to a potential inadvertent divergent change.